src/main/java/uk/co/jemos/podam/api/AbstractRandomDataProviderStrategy.java

```diff
@@ -542,8 +542,17 @@ public RandomDataProviderStrategy removeAttributeStrategy(
 	public AttributeStrategy<?> getStrategyForAnnotation(
 			final Class<? extends Annotation> annotationClass) {
 
-		return attributeStrategies.get(annotationClass);
+		if (attributeStrategies.containsKey(annotationClass)) {
+			return attributeStrategies.get(annotationClass);
+		} else {
+			for (Entry<Class<? extends Annotation>, AttributeStrategy<?>> attributeStrategy: attributeStrategies.entrySet()) {
+				if (attributeStrategy.getKey().isAssignableFrom(annotationClass)) {
+					return attributeStrategy.getValue();
+				}
+			}
+		}
 
+		return null;

```

Comment:
This is something what will make Sonar unhappy.

Reply:
you suggestion was applied